app/init/launch.ts

```diff
@@ -252,3 +257,17 @@ export const getLaunchPropsFromNotification = async (notification: NotificationW
 
     return launchProps;
 };
+
+async function cleanupEphemeralPosts() {
+    const servers = await getAllServers();
+
+    for (const server of servers) {
+        const database = DatabaseManager.serverDatabases[server.url]?.database;
+        if (!database) {
+            continue;
+        }
+
+        const postsPromise = queryPostsByType(database, 'system_ephemeral').fetch();
+        postsPromise.then((posts) => posts.forEach((post) => removePost(server.url, post)));
```

Comment:
Any strong reason to use the `.then` approach instead of using the `await` approach? The function is already asynchronous, and it would keep the code cleaner.

Reply:
Since we're not awaiting this function, I thought to remove `await Promise.all(...)` inside the function as well, thus changing `servers.map()` to a `for of` loop. But the linter panicked saying that `await` was not allowed inside a for loop.

Maybe I can use `servers.forEach()` instead and keep the `await` approach, or should I just keep the [old design](https://github.com/mattermost/mattermost-mobile/blob/a158d99d2c95e9616bd24d1d2d3c6b98b6154bea/app/init/launch.ts#L261-L275) and not bother removing `await Promise.all(...)`?

Reply:
The reason for the lint rule is that in most cases, you don't want to wait for one iteration of the loop in order to do the next. You want to run them all, and let the OS handle the potential concurrency. But I would say that in this case, specially after what we discussed about wanting to delete the posts one by one, it may be safer to do it with as little concurrency as possible (also taking in consideration they are a limited number of posts).

So... I would say keep the for, and just add a disable for that rule on that line.

What do you think?